src/main/java/io/lettuce/core/internal/Futures.java

```diff
@@ -32,10 +32,11 @@ public static CompletableFuture<Void> allOf(Collection<? extends CompletionStage
 
         LettuceAssert.notNull(stages, "Futures must not be null");
 
-        CompletableFuture[] futures = new CompletableFuture[stages.size()];
+        CompletionStage[] copies = stages.toArray(new CompletionStage[0]);
+        CompletableFuture[] futures = new CompletableFuture[copies.length];
 
```

Comment:
Why need to create a copy array if you only need to get the size?

Reply:
the issue stated that this method throws ArrayOutOfIndexException. @tishun mentioned that that is because of the concurrent modification of the 'stages' and I agreed. So I created copy array and used in the for loop.

```
for (CompletionStage<?> stage : copies) { //use copies instead of stages 
    futures[index++] = stage.toCompletableFuture();
}
```

Reply:
Ah, I see, thanks.